storage/memory/memory.go

```diff
@@ -463,27 +515,28 @@ func (m *memory) PredicatesForObject(ctx context.Context, o *triple.Object, lo *
 	if prds == nil {
 		return fmt.Errorf("cannot provide an empty channel")
 	}
+
 	oUUID := UUIDToByteString(o.UUID())
 	m.rwmu.RLock()
 	defer m.rwmu.RUnlock()
 	defer close(prds)
 
 	if lo.LatestAnchor {
-		lastTA := make(map[string]*time.Time)
-		trps := make(map[string]*triple.Triple)
-		for _, t := range m.idxO[oUUID] {
-			p := t.Predicate()
-			ppUUID := p.PartialUUID().String()
-			if p.Type() == predicate.Temporal {
-				ta, err := p.TimeAnchor()
-				if err != nil {
-					return err
-				}
-				if lta := lastTA[ppUUID]; lta == nil || ta.Sub(*lta) > 0 {
-					trps[ppUUID] = t
-					lastTA[ppUUID] = ta
-				}
-			}
+		if lo.FilterOptions != nil {
+			return fmt.Errorf("cannot have LatestAnchor and FilterOptions used at the same time inside lookup options")
+		}
+		lo.FilterOptions = &storage.FilteringOptions{
+			Operation: "latest",
+			Field:     "predicate",
+		}
+		defer func() {
+			lo.FilterOptions = (*storage.FilteringOptions)(nil)
+		}()
+	}
+	if lo.FilterOptions != nil {
+		trps, err := executeFilter(m.idxO[oUUID], lo.FilterOptions)
+		if err != nil {
+			return err
 		}
```

Comment:
I think this pattern here could be changed to 

``` go
  localFilterOptions := lo.FilterOptions
  if lo.LatestAnchor {
    if lo.FilterOptions != nil {
      return fmt.Error()
    }
  localFilterOptions = storage.FilteringOptions { Operation: "latest", Field: "predicate"}
  }
if !localFilterOptions.IsEmpty(){
...

```
WDYT?
I think the advantages here are that we are avoiding using defer and that the temp filter options is more explicit and can be stack variable

Reply:
I like your suggestion, as it avoids `defer` by elegantly leveraging a stack variable. But I did it the other way so we can keep `FilterOptions` more separated in code from `LatestAnchor`, and then at the moment of deprecating `LatestAnchor` (that I believe will be soon, as commented in the previous comment) we can directly delete the block of code related to it, without any refactoring.

This way, I am inclined to let the code as it is for now. But if you still prefer the other approach I can always change it to be as you suggested, just lmk.